### PR TITLE
Run clippy

### PR DESCRIPTION
**Pull request by sinkuu**

### CODE REVIEW

File: src/checkstyle.rs [hunk 1]
```diff
@@ -45,18 +45,15 @@ pub fn output_checkstyle_file<T>(mut writer: T,
...
-                }
+            if let DiffLine::Expected(ref str) = line {
+                let message = xml_escape_str(str);
+                try!(write!(writer,
+                            "<error line=\"{}\" severity=\"warning\" message=\"Should be `{}`\" \
+                             />",
+                            mismatch.line_number,
+                            message));
+            } else {
+                // Do nothing with context and expected.
```
Comment (nrc) [inline src/checkstyle.rs]: Could remove the else block and have the comment outside the if let

File: src/imports.rs [hunk 2]
```diff
@@ -162,8 +162,7 @@ impl<'a> FmtVisitor<'a> {
         // Find the location immediately before the first use item in the run. This must not lie
         // before the current `self.last_pos`
         let pos_before_first_use_item = use_items.first()
-            .map(|p_i| cmp::max(self.last_pos, p_i.span.lo))
-            .unwrap_or(self.last_pos);
+            .map_or(self.last_pos, |p_i| cmp::max(self.last_pos, p_i.span.lo));
```
Comment (nrc) [inline src/imports.rs]: This seems a bad change - it makes the code less clear IMO

File: src/issues.rs [hunk 3]
```diff
@@ -23,8 +23,7 @@ const FIX_ME_CHARS: &'static [char] = &['F', 'I', 'X', 'M', 'E'];
 impl ReportTactic {
     fn is_enabled(&self) -> bool {
         match *self {
-            ReportTactic::Always => true,
-            ReportTactic::Unnumbered => true,
+            ReportTactic::Always | ReportTactic::Unnumbered => true,
             ReportTactic::Never => false,
         }
```
Comment (nrc) [inline src/issues.rs]: This whole function could be just `self != ReportTactic::Never`

**Comment by nrc** [comment]
A few comments inline, but generally looks good, thanks!

**Comment by sinkuu** [comment]
Updated.

**Comment by nrc** [comment]
Thank you!

**Comment by nrc** [comment]
I think `ReportTactic` needs to derive `PartialEq` and `Eq`

**Comment by sinkuu** [comment]
Updated.

**Comment by nrc** [comment]
Thanks for updating, it passes tests now, but needs a rebase.

**Comment by sinkuu** [comment]
rebased.

**Comment by nrc** [comment]
Thanks!

**Pull request merged**